### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by rociac**
Looking forward for your feedback!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,3 +1,49 @@
...
   has_secure_password
+  class << self
+    def digest(string)
+      cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+
+      BCrypt::Password.create(string, cost: cost)
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
```
Comment (efrapp) [inline app/models/user.rb:27]: You missed step 5.2 in the Odin project, about encrypting the token.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,11 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'static_pages#home'
+  get 'microposts/new'
+  post 'microposts/new', to: 'microposts#create'
+  get 'microposts/index'
```
Comment (efrapp) [inline config/routes.rb]: It is better to use the `:resources` method.

File: app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @microposts = Micropost.all
+  end
+
+  private
+
+  def restrict
+    redirect_to login_path unless logged_in?
+  end
+
+  def user_params
```
Comment (efrapp) [inline app/controllers/microposts_controller.rb]: Maybe a more accurate name should be `micropost_params`

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+class MicropostsController < ApplicationController
+  before_action :restrict, only: %i[new create]
+
+  def new
+    @micropost = Micropost.new
+  end
+
+  def create
+    @micropost = Micropost.new(user_params)
+    @micropost.user = current_user
```
Comment (efrapp) [inline app/controllers/microposts_controller.rb]: These lines could be refactored using the `build` method from the association between users and microposts.

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,48 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  helper_method :logged_in?
+
+  def log_in(user)
+    session[:user_id] = user.id
+  end
```
Comment (efrapp) [inline app/controllers/application_controller.rb:9]: For this method, you should generate the remember token and update the signed-in user with that token.

**Review by efrapp — CHANGES_REQUESTED** [review]
### Hi @rociac and @onedebos 👋 
You are doing a good job so far. Please see my suggestion bellow.
Also, I wasn't able to test your app in the browser because you have some errors related to Sass, please fix those errors and send a new code review with the app completely functional.

**Required changes**
- [X] Please check this issue: #2
- [X] Check this issue also: #3. There are a lot of Sass undefined variables you are using in your project. (Maybe this is related to the issue #2)
- [ ] It uses the #current_user=(user) method to set user

**Status:** `Changes required`

File: app/models/user.rb (hunk 1 above)
Comment (rociac) [reply app/models/user.rb:27]: Hi, thank you for reviewing our code. We did encrypt the password using our digest method declared above inside class << self it uses the remember_token and the authenticated method to encrypt the password.

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (rociac) [reply app/controllers/application_controller.rb:9]: We did generate the remember token but it's split in different methods, we use the remember method and the current_user? method to achieve that.

File: config/routes.rb (hunk 2 above)
Comment (SunnyAsar) [reply config/routes.rb]: Hi, please update this as suggested previously, use the resources with only flag to specify the methods you want.

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
# Hi @rociac  and @onedebos 👋 
- your code looks good, and i see some updates from the previous review. There are still a few left.
- I'm unable to use the app generally please look into the sass issue.
Screenshot 2019-10-09 at 11 26 43 AM
### please fix this so we can move forward.
**Request another review when everything is in place**

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,12 @@
...
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root   'static_pages#home'
+  get    '/login',            to: 'sessions#new'
+  post   '/login',            to: 'sessions#create'
+  delete '/logout',           to: 'sessions#destroy'
+
+
+  resources :users, only: [:new, :create]
+  resources :microposts, only: [:new, :create, :index]
```
Comment (efrapp) [inline config/routes.rb:11]: Nice guys! 👍

**Review by efrapp — CHANGES_REQUESTED** [review]
### Great job team! 👷

Thanks for solving the problem with Sass, I was able to review the app without problems.
Please address the following suggestions:

- [ ] As a guest you can see only post's text: you need to allow to check all the posts even is no user is signed-in but without the author's name.
- [ ] It uses the #current_user=(user) method to set user
- [ ] the author is displayed only if a user is signed in (uses ApplicationController#signed_in? helper method)

**Status:** `Changes required`

**Review by Oghenebrume50 — APPROVED** [review]
looks good Team, all the changes work fine **This is approved**

**Pull request merged**